helm-chart/kuberay-operator/values.yaml

```diff
@@ -35,6 +35,16 @@ resources:
   #   cpu: 100m
   #   memory: 512Mi
 
+logging:
+  # Log encoder to use for stdout (one of 'json' or 'console', default is 'console')
+  stdoutEncoder: ""
```

Comment:
```yaml
logging:
  # Log encoder to use for stdout (one of 'json' or 'console', default is 'console')
  stdoutEncoder: ""
  # Log encoder to use for file logging (one of 'json' or 'console', default is 'json')
  fileEncoder: ""
  # Directory for kuberay-operator log file
  baseDir: "/tmp/kuberay"
  # File name for kuberay-operator log file
  fileName: "operator.log"
```

The STDOUT is in `json` format instead of `console`. I used `kubectl logs` to retrieve [these log messages](https://gist.github.com/kevin85421/53d55b7eeb6820940c55f8ab55b76e23).

Reply:
Good catch -- this is because the encoder defaults to `json` when Development mode is disabled. 
```
// Encoder configures how Zap will encode the output.  Defaults to
// console when Development is true and JSON otherwise
Encoder zapcore.Encoder
```

I think it would actually be better to adopt controller-runtime defaults and use stdout encoder as json. If you want console logging you can use `--zap-devel` or `--zap-encoder=console`. WDYT?

Reply:
I don't have a very strong opinion about using `console` or `json` for STDOUT by default, but @marton-bod's point also makes sense to me.

"Console logging works best if it's human-readable when you perform kubectl logs for debugging for example, whereas the log file is often better to be json-structured so that it can be efficiently processed by a log server like Splunk."

Reply:
In general I agree too. But many logging systems just read stout (like fluentbit) and using console encoding would be suboptimal in those cases. 

Also, there's probably a good reason controller runtime uses json encoding by default. Most projects I've seen using controller runtime use json for stout. I'll try to dig up more context on this decision to see if there any valuable insights there

Reply:
@marton-bod thinking about this more -- I think we should also default stdout to use JSON encoding, this is because many logging systems just read stdout of containers so this is probably a better default. Console logging can still be enabled with `--zap-encoder=console` or `--zap-devel`. What do you think?

Reply:
> Maybe it'd be easier-to-understand and more symmetrical to have another flag --stdout-encoder which sets the stdout encoder to json/console?

SGTM, I'll update

Reply:
Going to name it `--log-stdout-encoder` for consistency with existing falgs